frontend/src/queries/schema/schema-general.ts

```diff
@@ -1803,12 +1804,24 @@ export interface ExperimentMetric {
     metric_type: ExperimentMetricType
     filterTestAccounts?: boolean
     inverse?: boolean
-    metric_config: ExperimentEventMetricConfig | ExperimentDataWarehouseMetricConfig
+    metric_config: ExperimentEventMetricConfig | ExperimentActionMetricConfig | ExperimentDataWarehouseMetricConfig
 }
 
 export interface ExperimentEventMetricConfig {
     kind: NodeKind.ExperimentEventMetricConfig
     event: string
+    name?: string
+    math?: ExperimentMetricMath
+    math_hogql?: string
+    math_property?: string
+    /** Properties configurable in the interface */
+    properties?: AnyPropertyFilter[]
+}
+
+export interface ExperimentActionMetricConfig {
```

Comment:
One thought: We could remove `event` / `action` from these two schemas, and add `id` and `type: 'events' | 'action`. It will match what we get from the form then. We will get away with one schema and simplify `metricConfigToFilter` logic. Not a strong opinion, but it will be less code.

Reply:
@andehen Potentially! I think I'd like to keep them separate for now, as there's some value in having them explicitly defined as such.